tools/packaging/common/istio-iptables.sh

```diff
@@ -59,31 +60,58 @@ function dump {
     ip6tables-save
 }
 
-#
-#  Function check if argument is a valid ipv4 or ipv6 address
-#
 function isValidIP() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   if isIPv4 "${1}"; then
       true
-   elif [ "$1" != "${1#*:[0-9a-fA-F]}" ]; then
+   elif isIPv6 "${1}"; then
       true
    else
       false
    fi
 }
-
 #
 # Function return true if agrument is a valid ipv4 address
 #
 function isIPv4() {
-   if [ "$1" != "${1#*[0-9].[0-9].[0-9].[0-9]}" ]; then
+   local ipv4regexp="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$"
+   if [[ ${1} =~ ${ipv4regexp} ]]; then
       true
    else
       false
   fi
 }
-
-trap dump EXIT
+#
+# Function return true if agrument is a valid ipv6 address
+#
+function isIPv6() {
+  local ipv6section="^[0-9a-fA-F]{1,4}$"
+  addr="$1"
+  number_of_parts=0
+  number_of_skip=0
+  IFS=':' read -r -a addr <<< "$1"
+  for part in "${addr[@]}"; do
+    # check to not exceed number of parts in ipv6 address
+    if [[ ${number_of_parts} -ge 8 ]]; then
+        return 1
+    fi
+    if [[ ${number_of_parts} -eq 0 ]] && ! [[ ${part} =~ ${ipv6section} ]]; then
+        return 1
```

Comment:
return 0

Reply:
this is an error condition it should return 1, could you clarify why it should return 0?

Reply:
my bad... I didn't recall that bash function return 1 evaluates to false in the calling if boolean check. (bash is not fun)